hathor/transaction/transaction.py

```diff
@@ -274,9 +275,12 @@ def verify_basic(self, skip_block_weight_verification: bool = False) -> None:
 
     def verify_checkpoint(self, checkpoints: List[Checkpoint]) -> None:
         assert self.storage is not None
+        if self.is_genesis:
+            return
         meta = self.get_metadata()
+        get_partially_validated = partial(self.storage.get_transaction, allow_partially_valid=True)
```

Comment:
Why are we allowing partially validated transactions in this verification?

Reply:
this is inside `validate_checkpoint`, which is a validation that is not "full", it should be able to run for a checkpoint that has not yet connected to the genesis, so we have to allow partially valid here